cmd/kube-apiserver/app/server.go

```diff
@@ -301,6 +302,14 @@ func CreateKubeAPIServerConfig(s *options.ServerRunOptions, nodeTunneler tunnele
 		return nil, nil, nil, nil, nil, err
 	}
 
+	// Validate admission plugins dependency.
+	for _, pluginName := range s.Admission.PluginNames {
+		if strings.Contains(pluginName, "AdmissionWebhook") &&
+			!storageFactory.APIResourceConfigSource.AllResourcesForVersionEnabled(admissionregistration_v1beta1.SchemeGroupVersion) {
```

Comment:
This only helps kube-apiserver. Please take a look at #56859, if you think that's reasonable, you can rely on 56859 and remove your last commit.

Reply:
sure, this is no need now.